client/app/components/queries/QueryEditor/AutoLimitCheckbox.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useCallback } from "react";
+import PropTypes from "prop-types";
+import "@/redash-font/style.less";
+import recordEvent from "@/services/recordEvent";
+import { Checkbox } from "antd";
+
+export default function AutoLimitCheckbox({ available, checked, onChange }) {
+  const handleClick = useCallback(() => {
+    recordEvent("checkbox_auto_limit", "screen", "query_editor", { state: !checked });
+    onChange(!checked);
+  }, [checked, onChange]);
+
+  return (
+    <Checkbox
+      className="query-editor-controls-checkbox m-l-5"
+      disabled={!available}
+      onClick={handleClick}
+      checked={available && checked}>
+      <label style={{ fontSize: 9 }}>Limit 1000</label>
```

Comment:
Let's use the default font size + make sure it's aligned vertically to the middle of the buttons.

![CleanShot 2020-08-19 at 13 24 47](https://user-images.githubusercontent.com/71468/90623600-61268780-e21f-11ea-946b-50a796b11d48.png)

(I'm not sure if it's really in the middle in the screenshot, just added for illustration :-) ).

Reply:
Agreed with Arik

```suggestion
      Limit 1000
```

---
Also for me this is currently in the base line:
<img width="445" alt="Screen Shot 2020-08-19 at 13 30 59" src="https://user-images.githubusercontent.com/3356951/90663580-3be64900-e220-11ea-8f77-41879333a5a4.png">

You can add this to the styling you have below (and remove the `.m-l-5` class):

```css
.query-editor-controls-checkbox {
  /* ... */
  margin: auto 5px;
}
```
After that:
<img width="397" alt="Screen Shot 2020-08-19 at 13 36 59" src="https://user-images.githubusercontent.com/3356951/90664296-2c1b3480-e221-11ea-8060-5e821cbd61ad.png">

Reply:
Why limit default rows? why not limit by userself to edit rows? Like limit 2000 or limit anyrows?

Reply:
Just use `align-self: center` rule for the element that is a direct child of `.query-editor-controls`, and remove all margins/padding at all

Reply:
For me it works both with `align-self: center` or the `margin`. I was just leading towards the margin because it has a `.m-l-5` that can be removed if the rule is in `margin`. The parent display is already flex anyway 🤷‍♂️.